optimade/server/middleware.py

```diff
@@ -197,6 +192,8 @@ def handle_api_hint(api_hint: List[str]) -> Union[None, str]:
             # Go to /v<MAJOR>, since this should point to the latest available
             return BASE_URL_PREFIXES["major"]
 
+        return None
```

Comment:
I do not see the point of adding a return None here. 
There does not seem to be a way to reach it.
And the value BASE_URL_PREFIXES["major"] seems to be a reasonable fallback value. 
So, could you not remove the last if statement instead?  
Ideally, you would also give a warning that a different API version has been used to handle the request.

Reply:
Agree on removing `return None` as it should be unreachable, this is one where I've just tried to placate mypy.

Instead I've reorganized it to be (shorthand)

```
if major_api_hint <= major_implementation:
   return hint

raise exc
```

I don't think we want to change the fallback value though (i.e., we should still raise an error if someone hints for v2).

Reply:
That is also a good solution.